07_errors/patrickmarabeas/syncStore.go

```diff
@@ -13,19 +13,23 @@ func (store *SyncStore) Create(puppy Puppy) (int, error) {
 		return -1, NewError(NegativeValue)
 	}
 
+	store.Lock()
 	puppy.ID = store.uuid
 	store.Store(puppy.ID, puppy)
 	store.uuid++
+	store.Unlock()
```

Comment:
The usual Go pattern is to put this right after the `Lock()` with `defer`:
```
store.Lock()
defer store.Unlock()
```

This way if you later add an early return in the middle for error handling, you do not forget to unlock.